Source/WarpX.cpp

```diff
@@ -1158,14 +1176,13 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     current_fp[lev][1] = std::make_unique<MultiFab>(amrex::convert(ba,jy_nodal_flag),dm,ncomps,ngJ,tag("current_fp[y]"));
     current_fp[lev][2] = std::make_unique<MultiFab>(amrex::convert(ba,jz_nodal_flag),dm,ncomps,ngJ,tag("current_fp[z]"));
 
```

Comment:
I added this in similarity to
https://github.com/ECP-WarpX/WarpX/blob/b15ee7b890b51d4ec15e36d8e41465b49dcfa058/Source/WarpX.cpp#L1167-L1173

I think the initial reason for adding this extra ghost cell with momentum-conserving gathering was that with the _old interpolation functions_ (which did not have padding with zeros around the ghost cells of the input staggered arrays and which performed always linear interpolation only) we needed one extra point when interpolating from a staggered field to a nodal field, in order to properly fill the last nodal point in the output array (which would be otherwise filled with junk, like `NaN` or similar, without padding):

```
--o-----o-----o-----o
-----x-----x-----x:::::x  (this was the extra ghost point)
```

However, with the _new interpolation functions_, where we now have padding with zeros around the ghost cells of the input staggered arrays, we probably do not need this anymore.

And I'm thinking now that it is probably safe to remove the extra ghost cells from the `fp` MultiFab as well (not only from the `avg_fp` MultiFabs).

So maybe I could try to avoid adding them here for the `avg_fp` MultiFabs as well as try to remove them from the `fp` MultiFabs in a separate PR. Given that we are talking about the last ghost points, they should not matter at all in the end for the subsequent field gathering operations (at least with PSATD for sure, where we have many ghost cells which span a region larger than (half of) the support of the shape factors). What do you think?

Reply:
This would also require us to modify the code here:
https://github.com/ECP-WarpX/WarpX/blob/b15ee7b890b51d4ec15e36d8e41465b49dcfa058/Source/Evolve/WarpXEvolve.cpp#L116-L120

If we try to proceed as I described above, we would need to pass `0` instead of `guard_cells.ng_Extra` in the function calls here, otherwise we will be requesting ghost cells exchange on a number of ghost cells larger than the ones that are actually allocated.

I will try all of this in a separate PR.

Reply:
Thanks to #1758 and https://github.com/ECP-WarpX/WarpX/pull/1544/commits/2678684f3782be44ad7a8995f577ab0353ccea98, we do not have extra ghost cells anymore.